src/appimagetool.c

```diff
@@ -457,6 +459,41 @@ bool readFile(char* filename, int* size, char** buffer) {
     return TRUE;
 }
 
+/* run a command outside the current appimage, block environs like LD_LIBRARY_PATH */
+int run_external(const char *filename, char *const argv []) {
+    int pid = fork();
+    if (pid == -1) {
+        g_print("run_external: fork failed");
+        // fork failed
+        return -1;
+    } else if (pid == 0) {
+        // blocks env defined in resources/AppRun
+        unsetenv("LD_LIBRARY_PATH");
+        unsetenv("PYTHONPATH");
+        unsetenv("XDG_DATA_DIRS");
+        unsetenv("PERLLIB");
+        unsetenv("GSETTINGS_SCHEMA_DIR");
+        unsetenv("QT_PLUGIN_PATH");
+        // runs command
+        execve(filename, argv, environ);
+        // execve(3) returns, indicating error
+        g_print("run_external: subprocess execve(3) got error %s", g_strerror(errno));
+        exit(1);
+    } else {
+        int wstatus;
+        if (waitpid(pid, &wstatus, 0) == -1) {
+            g_print("run_external: wait failed");
+            return -1;
+        }
+        if (WIFEXITED(wstatus) && (WEXITSTATUS(wstatus) == 0)) {
+            return 0;
+        } else {
+            g_print("run_external: subprocess exited with non-0 status");
```

Comment:
Please print and return the exact exit code here. This is not necessarily helpful to the user who must debug the subprocess failure.